src/spikeinterface/core/analyzer_extension_core.py

```diff
@@ -410,11 +558,21 @@ def _compute_and_append_from_waveforms(self, operators):
 
     @property
     def nbefore(self):
-        return self.params["nbefore"]
+        waveforms_extension = self.sorting_analyzer.get_extension("waveforms")
+        if waveforms_extension is not None:
+            nbefore = waveforms_extension.nbefore
+        else:
+            nbefore = int(self.params["ms_before"] * self.sorting_analyzer.sampling_frequency / 1000.0)
+        return nbefore
```

Comment:
bad idea.
The wavefortms extesion could be deleted in bteween!
The params must captured on the run.

Reply:
The set_params() can only save params that are in the init, nothing more. Otherwise, we have issues when reloading extensions. Let's discuss that

Reply:
This should be set in the run

Reply:
Ok, then I'll let the two of you do that. But this means that if the extension is not runed, this params can not be obtained

Reply:
done